cmd/deploy_test.go

```diff
@@ -1,182 +1,261 @@
 package cmd
 
 import (
-	"context"
-	"os"
+	"fmt"
 	"testing"
 
-	"github.com/ory/viper"
-	"k8s.io/utils/pointer"
 	fn "knative.dev/kn-plugin-func"
 	"knative.dev/kn-plugin-func/mock"
 	. "knative.dev/kn-plugin-func/testing"
 )
 
-func Test_runDeploy(t *testing.T) {
-	tests := []struct {
-		name                 string
-		gitURL               string
-		gitBranch            string
-		gitDir               string
-		buildType            string
-		funcFile             string
-		expectFileURL        *string
-		expectFileBranch     *string
-		expectFileContextDir *string
-		expectCallURL        *string
-		expectCallBranch     *string
-		expectCallContextDir *string
-		errString            string
-	}{
-		{
-			name:      "Git arguments don't get saved to func.yaml but are used in the pipeline invocation",
-			gitURL:    "git@github.com:knative-sandbox/kn-plugin-func.git",
-			gitBranch: "main",
-			gitDir:    "func",
-			buildType: fn.BuildTypeGit,
-			funcFile: `name: test-func
-runtime: go
-created: 2009-11-10 23:00:00`,
-			expectCallURL:        pointer.StringPtr("git@github.com:knative-sandbox/kn-plugin-func.git"),
-			expectCallBranch:     pointer.StringPtr("main"),
-			expectCallContextDir: pointer.StringPtr("func"),
-		},
-		{
-			name:      "Git url gets split when in the format url#branch",
-			gitURL:    "git@github.com:knative-sandbox/kn-plugin-func.git#main",
-			gitDir:    "func",
-			buildType: fn.BuildTypeGit,
-			funcFile: `name: test-func
-runtime: go
-created: 2009-11-10 23:00:00`,
-			expectCallURL:        pointer.StringPtr("git@github.com:knative-sandbox/kn-plugin-func.git"),
-			expectCallBranch:     pointer.StringPtr("main"),
-			expectCallContextDir: pointer.StringPtr("func"),
-		},
-		{
-			name:      "Git arguments override func.yaml but don't get saved",
-			gitURL:    "git@github.com:knative-sandbox/kn-plugin-func.git",
-			gitBranch: "main",
-			gitDir:    "func",
-			funcFile: `name: test-func
```

Comment:
This PR updates tests to use the Client API rather than write func.yaml directly, as this is more likely to be backwards compatible (the exact format of func.yaml is an implementation detail of the client package not cmd, so tests relying on its structure should probably only exist there)